djes/signals.py

```diff
@@ -0,0 +1,10 @@
+from django.db.models.signals import post_delete
+from django.dispatch import receiver
+
+from .models import Indexable
+
+
+@receiver(post_delete)
+def delete_es_index_on_delete(sender, instance, **kwargs):
+    if isinstance(instance, Indexable):
```

Comment:
Makes sense to me. 

As for the larger signal-vs-method debate, I agree that our convention should be to minimize signal usage, maybe we just restrict to simple operations on delete (and maybe save() or similar).